src/main/java/com/google/api/codegen/discovery/Document.java

```diff
@@ -133,6 +147,22 @@ public Schema dereferenceSchema(Schema schema) {
   }
 
   /** @return the auth instructions URL. */
+  @JsonIgnore @Nullable private Node parent;
+
+  public Node parent() {
+    return parent;
+  }
+
+  // Package private for use by other Node objects.
+  void setParent(Node parent) {
```

Comment:
Why have a setter? If the issue was null nodes, you can define it as:

```
@JsonProperty("parent")
@Nullable
public abstract Node parent();
```

Same suggestion for below.